lang/cs/Org.Apache.REEF.Common/metrics/Api/IImmutableMetricCommonFunctions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Utilities;
+
+namespace Org.Apache.REEF.Common.Metrics.Api
+{
+    /// <summary>
+    /// Contains functions whose implementation are common to all the metrics. 
+    /// All these functions are implemented once via <see cref="ImmutableMetricsImpl"/>.
+    /// </summary>
+    public interface IImmutableMetricCommonFunctions : IMetricTagCommonFunctions
+    {
+        /// <summary>
+        /// Long Value of the metric. Immutable metrics of 
+        /// type integrals, byte, bool are all type casted to long before storing
+        /// them as immutable metrics. For a given instance of this interface, either this property 
+        /// or <see cref="NumericValue"/> returns a valid value.
+        /// </summary>
+        Optional<long> LongValue { get; }
```

Comment:
We might also want to take this to the dev list, as have plenty places where we use `Optional`